### ISSUE TITLE
Introduce full seperation between secrets & configuration

### ISSUE THREAD

**Issue by tomkerkhove** [arcus-azure/arcus#69]
Since https://github.com/dotnet/extensions/issues/3088 won't happen soon and @stijnmoreels & myself like the idea; let's add it to Arcus.

An example is the introduction of `ConfigureSecretStore`:
```csharp
public static IHostBuilder CreateHostBuilder(string[] args) =>
    Host.CreateDefaultBuilder(args)
        .ConfigureAppConfiguration((context, config) =>
        {
            config.AddEnvironmentVariables();
        })
        .ConfigureSecretStore((context, config, secretStores) =>
        {
            if (context.HostingEnvironment.IsProduction())
            {
                var keyVaultName = configuration["KeyVaultName"];
                secretStores.AddAzureKeyVault(keyVaultName);
            }
            else
            {
                secretStores.AddConfiguration();
            }
        })
        .ConfigureWebHostDefaults(webBuilder =>
        {
            webBuilder.UseStartup<Startup>();
        });
```

Secret stores can be configured via ConfigureSecretStore which feels similar to ConfigureAppConfiguration allowing you to add providers and gives access to configuration.

In our case, secret stores are what we call `ISecretProvider`.

**Comment by tomkerkhove** [comment]
Before we get started, can you check what we would need to change @stijnmoreels?

I think we should be able to move this to Arcus Security as this is runtime independent, right?

**Comment by stijnmoreels** [comment]
Yes, it seems logical to have it in the same namespace/assembly as the Azure Key Vault stuff too.

**Comment by stijnmoreels** [comment]
>In our case, secret stores are what we call ISecretProvider.

Will we register directly this interface? Or maybe provide a separate `ISecretStore` for this?

**Comment by stijnmoreels** [comment]
> > In our case, secret stores are what we call ISecretProvider.
> 
Will we register directly this interface? Or maybe provide a separate `ISecretStore` for this?

**Comment by tomkerkhove** [comment]
I'm not sure yet, what would the use be of it? You'd want to rebrand `ISecretProvider` as a provider for various secret sources, such as Azure Key Vault,  and `ISecretStore` to just fetch it then; regardless of configured stores?

**Comment by stijnmoreels** [comment]
Hmm,m yeah, you might be right. I was thinking of that we would have a direct relation between the "provider" and the "store" but I actually there's no problem I think.
It would maybe even make it more simpler and reusable, even for older versions.

**Comment by stijnmoreels** [comment]
@tomkerkhove , can I start by making a PR on the security repo for this?
I was thinking about a separate package but referenced by others (like `Arcus.Security.Providers.KeyVault`) so it is included.

**Comment by tomkerkhove** [comment]
That sounds good to me, but I'd use `Arcus.Security.Secrets.Core` as that seperate package since we already have it and feels like best place, makes sense?
